src/AST-Core/RBValueNode.class.st

```diff
@@ -58,11 +47,7 @@ RBValueNode >> evaluateForContext: aContext [
 	"evaluate the AST taking variables from the context"
 	| methodNode |
 	
-	methodNode := RBMethodNode 
-		selector: #DoItIn:
-		arguments: { RBVariableNode named: 'ThisContext' } 
-		body: (RBReturnNode value: self) asSequenceNode.
-
+	methodNode := self asDoitIn.
```

Comment:
Will not it be better to introduce "asDoitIn: aContext" instead of no arg asDoitIn?
Looks like this place is the only sender.

Reply:
but you do not need the aContext there... and more important: you might not even have it at the point where you call the method. (as this just compiles a method, that method later can take the context as an argument)

Reply:
ah, ok. Yes, this is possible. I will do it in the  next iteration. and the name should maybe be asDoItForContext: